crates/sp-lightclient/src/lib.rs

```diff
@@ -363,6 +374,36 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
         Ok(())
     }
 
+    fn find_era_start_slot(
+        &self,
+        header: &Header,
+        era_duration: NumberOf<Header>,
+    ) -> Result<Slot, ImportError<Header>> {
+        // special case when the current header is one, then extract pre digest and return slot.
+        if header.number().is_one() {
+            let digests = extract_pre_digest(header)?;
+            return Ok(digests.slot);
+        }
+
+        // if the current header number is less than or equal to era duration, then pick the slot from One
+        let era_start_number = if *header.number() <= era_duration {
+            One::one()
+        } else {
+            // era start slot is slot at (current number - era duration) block
+            header
+                .number()
+                .checked_sub(&era_duration)
+                .expect("Safe to unwrap due to check above")
+        };
+
+        let era_start_header = self
+            .find_ancestor_of_header_at_number(*header.parent_hash(), era_start_number)
+            .ok_or_else(|| ImportError::MissingParent(*header.parent_hash()))?;
```

Comment:
Era start can be below archiving/pruning point and this will fail. I think you'll have to store it somewhere.

Reply:
responded here - https://github.com/subspace/subspace/pull/740#discussion_r938376577

Lets continue the discussion there :-D